### DISCUSSION TITLE
Need some examples for the extract  UQL command

### DISCUSSION THREAD

**Discussion by simonarbel** [grafana/grafana-infinity-datasource#565] [Q&A]
Hi

I need to use the "extract" command to extract a name from a long string.
for example, extracting the item in bold:  

/subscriptions/XXXX/resourceGroups/PAW/providers/Microsoft.Network/azureFirewalls/**AzureFirewall_VWAN**/providers/Microsoft.Insights/metrics/DataProcessed

Can anyone provide the right syntax for it?

Thanks

**Comment by CraigLager** [comment]
I'm finding the usage for extract confusing - did you find any good examples?

**Comment by yesoreyeram** [comment]
see if this helps https://github.com/yesoreyeram/uql/blob/8e11cfa0fab7307d1e6a4412625d448e5fcbe4d8/src/parser/project/project.test.ts#L5
